src/Cassandra/LocalDate.cs

```diff
@@ -45,47 +42,31 @@ public class LocalDate: IComparable<LocalDate>, IEquatable<LocalDate>
         internal LocalDate(uint days)
         {
             DaysSinceEpochCentered = days;
-            var d0 = Convert.ToInt64(days - DateCenter +  DaysFromYear0ToUnixEpoch);
-            var y400 = d0 / DaysIn400Years;
-            var d = d0 % DaysIn400Years;
-            var y100 = d / DaysIn100Years;
-            d %= DaysIn100Years;
-            var y4 = d / DaysIn4Years;
-            d %= DaysIn4Years;
-            //No risk of overflow
-            var year = Convert.ToInt32(y400*400 + y100*100 + y4*4);
-            var isLastMultipleOf4Leap = IsLeapYear(year);
-            if (d >= 0)
-            {
-                if (isLastMultipleOf4Leap)
-                {
-                    for (var i = 0; i < 3; i++)
-                    {
-                        var daysInYear = 365 + (i == 0 ? 1 : 0);
-                        if (d < daysInYear)
-                        {
-                            break;
-                        }
-                        d -= daysInYear;
-                        year++;
-                    }
-                }
-                else
-                {
-                    year += (int) Math.Floor(d/365D);
-                    d %= 365;
-                }
-                CalculateMonthDay(IsLeapYear(year), Convert.ToInt32(d));
-            }
-            else
-            {
-                year += (int)Math.Floor(d / 365D);
-                var isLeap = IsLeapYear(year);
-                //Days in the year - rest days to discount
-                var dayInYear = (isLeap ? 366 : 365) + Convert.ToInt32(d % 365);
-                CalculateMonthDay(isLeap, dayInYear);
-            }
-            Year = year;
+            InitializeFromDaysSinceEpoch(days - LocalDate.DateCenter);
+        }
+
+        /// <summary>
+        /// Port from https://github.com/HowardHinnant/date which is based on http://howardhinnant.github.io/date_algorithms.html
+        /// There's a good explanation of the algorithm over there.
+        /// </summary>
+        /// <param name="daysSinceEpoch">Days since Epoch (January 1st, 1970), can be negative.</param>
+        private void InitializeFromDaysSinceEpoch(long daysSinceEpoch)
+        {
+            daysSinceEpoch += 719468L;                                                                  // shift epoch from 1970-01-01 to 0000-03-01
+            var era = (daysSinceEpoch >= 0 ? daysSinceEpoch : daysSinceEpoch - 146096) / 146097;        // compute era
```

Comment:
We could explain in code comment that in this algorithm, an era is a 400 year period.

Reply:
done